### PR TITLE
Trac 2910 Cleaned up the save procedures; left readers intact

### PR DESCRIPTION
**Pull request by alf239**
- cleaned up the save procedures
- left readers intact
- added comments on our NetCDF peculiarities (could be done better)

### CODE REVIEW

File: atlas-data-storage/src/main/java/uk/ac/ebi/gxa/data/NetCDFDataCreator.java [hunk 1]
```diff
@@ -241,6 +241,13 @@ private void create(NetcdfFileWriteable netCdf) throws IOException {
                 "DEacc", DataType.CHAR,
                 new Dimension[]{designElementDimension, designElementLenDimension}
         );
+        // !!!KLUDGE!!! Here be dragons
+        //
+        // Actually, "GN" is a long arrayÑand it is read as a long array
```
Comment (olgamelnichuk) [inline atlas-data-storage/src/main/java/uk/ac/ebi/gxa/data/NetCDFDataCreator.java]: "Ñ" ??

Comment (alf239) [inline atlas-data-storage/src/main/java/uk/ac/ebi/gxa/data/NetCDFDataCreator.java]: ouch!

Comment (alf239) [inline atlas-data-storage/src/main/java/uk/ac/ebi/gxa/data/NetCDFDataCreator.java]: Fixed — and improved the description; before that, it didn't explain **why** we couldn't simply use `LONG`

**Comment by alf239** [comment]
Note: I hasn't created a proper unit test—thinking about it now. Still, it's harmless so far, and is already tested manually.

There's even a unit test which sort of tests the sequence—but, again, it's more like getting a fixed point rather than properly testing to crash.

**Pull request merged**